allennlp/models/constituency_parser.py

```diff
@@ -0,0 +1,355 @@
+            A list of the pos tags for the words in the sentence, if they
+            were either predicted or taken as input to the model.
+
+        Returns
+        -------
+        A nested dictionary, where each node contains the following keys:
+
+        label : ``str``
+            The constituency label of this subtree.
+        start : ``int``
+            The start index for this subtree.
+        end : ``int``
+            The exclusive end index for this subtree.
+        children : ``List[Dict]``
+            A list of subtrees containing the children of this node.
+
+        or alternatively, if the node is a leaf node, it will have the following keys:
+
+        label : ``str``
+            The constituency label of this subtree.
+        start : ``int``
+            The start index for this subtree.
+        end : ``int``
+            The exclusive end index for this subtree.
+        is_leaf : ``bool`` = True
+            A indicator to make identifying leaf nodes easier.
+        pos_tag : ``str``, optional.
+            Optionally the gold pos tag will be included, if a list of
+            gold (or predicted) pos tags are passed to this method.
+        """
+        def assemble_subtree(start: int, end: int):
+            if (start, end) in spans_to_labels:
+                label = spans_to_labels[(start, end)]
+            else:
+                label = None
+
+            # This node is a leaf.
+            if end - start == 1:
+                word = sentence[start]
+                tree = {"start": start, "end": end, "word": word, "is_leaf": True}
+                if label is not None:
+                    tree["label"] = label
+                if pos_tags is not None:
+                    tree["pos_tag"] = pos_tags[start]
+                return [tree]
+
+            argmax_split = start + 1
+            # Find the next largest subspan such that
+            # the left hand side is a constituent.
+            for split in range(end - 1, start, -1):
+                if (start, split) in spans_to_labels:
+                    argmax_split = split
+                    break
+
+            left_trees = assemble_subtree(start, argmax_split)
+            right_trees = assemble_subtree(argmax_split, end)
+            children = left_trees + right_trees
+            if label is not None:
+                children = [{"label": label, "children": children, "start": start, "end": end}]
+            return children
+
+        tree = assemble_subtree(0, len(sentence))
+        return tree[0]
+
+    @overrides
+    def get_metrics(self, reset: bool = False) -> Dict[str, float]:
+
+        all_metrics = {}
+        for metric_name, metric in self.metrics.items():
+            f1, precision, recall = metric.get_metric(reset) # pylint: disable=invalid-name
+            all_metrics[metric_name + "_f1"] = f1
+            all_metrics[metric_name + "_precision"] = precision
+            all_metrics[metric_name + "_recall"] = recall
```

Comment:
Do you want to show an average F1, or something?  Seems like this would get unwieldy pretty quickly.